### PR TITLE
Inject runbook url into alerts

### PR DESCRIPTION
**Pull request by lilic**
This is part of the alerting epic, which requires all critical alerts to have runbooks, this is done via the runbook_url field.

We can merge after https://github.com/openshift/runbooks/pull/9 this PR is merged, as right now two runbooks don't exist yet.

/hold

### CODE REVIEW

**Comment by lilic** [comment]
@hexfusion @marun  please take a look, thanks!

File: jsonnet/main.jsonnet [hunk 1]
```diff
@@ -1,12 +1,23 @@
 local etcdMixin = (import 'github.com/etcd-io/etcd/contrib/mixin/mixin.libsonnet');
 local openshiftRules = (import 'custom.libsonnet');
-local a = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
-local r = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
-local o = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
+
+local alertingRules = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
+local promRules = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
+local openshiftAlerts = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
```
Comment (paulfantom) [inline jsonnet/main.jsonnet:6]: out of curiosity why not put all this into one variable with:
```
rules = etcdMixin.prometheusAlerts.groups + etcdMixin.prometheusRules.groups + openshiftRules.prometheusRules.groups
```
?

You are in full control over the fact of those `groups` existence and this way you can simplify code a bit.

File: jsonnet/main.jsonnet [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+
 // Exclude rules that are either OpenShift specific or do not work for OpenShift.
 // List should be ordered.
-local rules = std.map(function(group) group { rules: std.filter(function(rule) !std.setMember(rule.alert, ['etcdHighNumberOfFailedGRPCRequests', 'etcdHighNumberOfLeaderChanges', 'etcdInsufficientMembers']), super.rules) }
-                      , a + r);
+local excludeRules = std.map(function(group) group { rules: std.filter(function(rule) !std.setMember(rule.alert, ['etcdHighNumberOfFailedGRPCRequests', 'etcdHighNumberOfLeaderChanges', 'etcdInsufficientMembers']), super.rules) }, alertingRules + promRules);
```
Comment (paulfantom) [inline jsonnet/main.jsonnet:10]:
```suggestion
local excludedAlerts = ['etcdHighNumberOfFailedGRPCRequests', 'etcdHighNumberOfLeaderChanges', 'etcdInsufficientMembers']
local excludeRules = std.map(
    function(group) group {
      rules: std.filter(
        function(rule) !std.setMember(rule.alert, excludedAlerts), super.rules
      )
    }, alertingRules + promRules
  );
```
nit :)

File: jsonnet/main.jsonnet (hunk 1 above)
Comment (paulfantom) [inline jsonnet/main.jsonnet:6]: nit: I think you can put all those under one variable and simplify further filtering. 
```suggestion
local alertingRules = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
local promRules = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
local rules = alertingRules + promRules + openshiftRules.prometheusRules.groups
```

Checking for group inclusion is necessary when you are importing code from another repository. In the case of the same repository (example `openshiftRules.prometheusRules.groups`) it is not so important as you are in full control.

File: jsonnet/main.jsonnet [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+
+// modifyRules injects runbook_url to all critical alerts on all rules.
+local modifyRules = std.map(function(group) group {
```
Comment (paulfantom) [inline jsonnet/main.jsonnet:13]: mega nit:
```suggestion
local modifiedRules = std.map(function(group) group {
```

`modifyRules` implies action, hence function, here we have data only.

File: jsonnet/main.jsonnet (hunk 1 above)
Comment (lilic) [reply jsonnet/main.jsonnet:6]: Thanks! We can't do this exactly as we need those separate and not in the same group, but sort of went with similar pattern.

Comment (paulfantom) [reply jsonnet/main.jsonnet]: I think I am missing why those need to be separated as all filters check first if entry is a recording rule or an alert :thinking:

Comment (lilic) [reply jsonnet/main.jsonnet]: The openshift rules we do not want to filter out, only exclude alerts from the upstream mixin.

Comment (lilic) [reply jsonnet/main.jsonnet]: Or did you mean to add promRules+alertingRules together?

Comment (paulfantom) [reply jsonnet/main.jsonnet]: All good, I need to re-learn how to read though :joy:

**Comment by lilic** [comment]
/hold cancel

@hexfusion please take a look, all runbooks now merged. 🎉

**Comment by lilic** [comment]
/retest

auth failure...

**Pull request merged**